soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -134,27 +134,42 @@ void dsp_restore_vector(void);
 
 void power_gate_entry(uint32_t core_id)
 {
-	struct lpsram_header *lpsheader =
-		(struct lpsram_header *) DT_REG_ADDR(DT_NODELABEL(sram1));
-
 	xthal_window_spill();
```

Comment:
I particularly don't like using HAL API when we have equivalent implemented. We could use our API the limitation is that we need save `a0` and do `call0` in assembler.

Reply:
Could you explain more precisely what you mean?

Reply:
I'm just saying that I prefer using Zephyr's native API (when avaialable) than the HAL API. That's said the API to spill window register on Zephyr has constraints different from the one in the HAL and I was explaining that.

Reply:
I prefer to keep it as it is. This PR is resolving issues with secondary cores power flows and I don't wanna do too much refactoring in it.

Reply:
that is reasonabe